cirq-ft/cirq_ft/algos/apply_gate_to_lth_target.py

```diff
@@ -77,7 +77,7 @@ def selection_registers(self) -> Tuple[infra.SelectionRegister, ...]:
 
     @cached_property
     def target_registers(self) -> Tuple[infra.Register, ...]:
-        total_iteration_size = np.product(
+        total_iteration_size = np.prod(
```

Comment:
for my edification: why this change?

Reply:
It fails mypy checks because `np.product` has been deprecated.